File: trunk/esper/src/test/java/net/esper/support/event/SupportEventTypeFactory.java
package net.esper.support.event;

import net.esper.event.EventType;

import java.util.Map;

public class SupportEventTypeFactory
{
    public static EventType createBeanType(Class clazz)
    {
        return SupportEventAdapterService.getService().addBeanType(clazz.getName(), clazz);
    }

    public static EventType createMapType(Map<String,Class> map)
    {
        return SupportEventAdapterService.getService().createAnonymousMapType(map);
    }
}

File: NEsper.Tests/support/event/SupportEventTypeFactory.cs
using System;

using net.esper.compat;
using net.esper.events;

namespace net.esper.support.events
{

    public class SupportEventTypeFactory
    {
        public static EventType CreateBeanType(Type clazz)
        {
            return SupportEventAdapterService.Service.AddBeanType(clazz.FullName, clazz);
        }

        public static EventType CreateMapType(EDictionary<String, Type> map)
        {
            return SupportEventAdapterService.Service.CreateAnonymousMapType(map);
        }
    }
}
